src/Symfony/Component/Routing/Loader/AnnotationDirectoryLoader.php

```diff
@@ -38,7 +38,15 @@ public function load($path, $type = null)
 
         $collection = new RouteCollection();
         $collection->addResource(new DirectoryResource($dir, '/\.php$/'));
-        $files = iterator_to_array(new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($dir), \RecursiveIteratorIterator::LEAVES_ONLY));
+        $files = iterator_to_array(new \RecursiveIteratorIterator(
+            new \RecursiveCallbackFilterIterator(
+                new \RecursiveDirectoryIterator($dir),
+                function (\SplFileInfo $current) {
+                    return '.' !== $current->getBasename()[0];
+                }
+            ),
+            \RecursiveIteratorIterator::LEAVES_ONLY
```

Comment:
`.` and `..` are already filtered out by `'.' !== $current->getBasename()[0];` :)

Not sure about following symlinks. This is not a problem we're solving atm, so I wouldn't add it until someones comes up with a use case